### PR TITLE
Float size2

### PR DESCRIPTION
**Pull request by sdancer**
updated pr

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
   permits callback using code to be written without indentation.
 - Fixed a bug where the warning for `todo` would not print the type of the code
   to complete.
+- Fixed float-size in bitstring
```
Comment (lpil) [inline CHANGELOG.md:10]: Could you explain what this means? Ideally a user can read this and know what new things they can do.

File: compiler-core/src/bit_string.rs [hunk 2]
```diff
@@ -234,15 +234,7 @@ fn type_options<T>(
...
-    // size cannot be used with float
-    match categories {
-        SegmentOptionCategories {
-            typ: Some(Float { .. }),
-            size: Some(opt),
-            ..
-        } => err(ErrorType::FloatWithSize, opt.location()),
-        _ => Ok(categories.segment_type()),
-    }
+    Ok(categories.segment_type())
```
Comment (lpil) [inline compiler-core/src/bit_string.rs:237]: Is FloatWithSize still used?

**Review by lpil — COMMENTED** [review]
Nice work!

Could we get runtime tests too to show it working? https://github.com/gleam-lang/gleam/blob/8970ea76e3daead73ab61dc68b9e1be8b37c9f9f/test/language/src/main.gleam#L1039

Thanks!

File: compiler-core/src/bit_string.rs (hunk 2 above)
Comment (sdancer) [reply compiler-core/src/bit_string.rs:237]: should be used in the future to validate static size of floats to 16/32/64 only, not used now

Comment (sdancer) [reply compiler-core/src/bit_string.rs:237]: actually will add that now(FloatWithSize), lacked enough understanding to add it two days ago, will also add the runtime tests

File: CHANGELOG.md (hunk 1 above)
Comment (lpil) [reply CHANGELOG.md:10]: Could you update this please? 🙏

File: compiler-core/src/bit_string.rs (hunk 2 above)
Comment (lpil) [reply compiler-core/src/bit_string.rs:264]: Nice, thanks!

**Review by lpil — APPROVED** [review]
Thank you!

**Comment by lpil** [comment]
Sorry, could you resolve the merge conflict please?

**Comment by sdancer** [comment]
nice

**Pull request closed without merging**